### ISSUE TITLE
Default Theme - Accessibility - Lack of visual focus indicator

### ISSUE THREAD

**Issue by dylanhthomas** [nopSolutions/nopCommerce#5022]
nopCommerce version: 4.3

## Steps to reproduce the problem:

1. Navigate to homepage of demo site: https://demo.nopcommerce.com/
2. Hit 'tab' repeatedly and observe the focused link (lower left of screen in Chrome)
 
## Result

There is no visual indicator of the focused item.

## Expected Results

Items which are focused via keyboard navigation should be indicated visually.

## Fix

1. Implement a default focus state that can be inherited.
2. Where there are custom :hover states, add :focus selectors so the mouse and keyboard experiences match.

### 1 - Default focus state

Following is added to:
_src/Presentation/Nop.Web/Themes/DefaultClean/Content/css/styles.css
src/Presentation/Nop.Web/Themes/DefaultClean/Content/css/styles-rtl.css_

```css
*:focus {
	outline: 3px dotted #333;
}
```

### 2 - Replicate :hover styles as :focus styles

Examples:

```css
.cart a:hover {}
```
becomes:
```css
.cart a:hover,
.cart a: focus {}
```

# WCAG Guideline

https://www.w3.org/WAI/WCAG21/quickref/?currentsidebar=%23col_overview#focus-visible
https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html

**Comment by dylanhthomas** [comment]
I'm working on this.

**Comment by dylanhthomas** [comment]
Fixed in PR #5028

**Comment by dylanhthomas** [comment]
@AndreiMaz BlueModus is a nopCommerce partner, and we focus on accessibility a lot, and would like to contribute to the project. Is this an area we can help? Is this a good way to do so?

**Comment by AndreiMaz** [comment]
@dylanhthomas Sure. Thanks a lot! You're more than welcome to contribute

**Comment by adamjohnson** [comment]
Dylan's issues / PR's have been on point from an accessibility standpoint. I'm fixing similar issues on a theme level but would love to see some of his PR's merged into the core.

FWIW, keyboard access (focus) is one of the [most important](https://accessibility.18f.gov/checklist/) things you can do for accessibility. Personally I like using the `Highlight` keyword for focus styles, eg:

```css
a:focus {
  outline: 2px solid Highlight;
}
```

**Comment by dylanhthomas** [comment]
Thanks @adamjohnson 
I'm not familiar with the "Highlight" keyword and some quick searching has me coming up empty. It sounds really useful, can you point me to some docs?

**Comment by adamjohnson** [comment]
Sure, here's where it's mentioned in [the spec](https://www.w3.org/TR/css-color-3/#highlight) and [the article](https://ghinda.net/article/mimic-native-focus-css/) where I found it. 👍

**Comment by ActionWebsites** [comment]
Sent from my Samsung Galaxy smartphone.
-------- Original message --------From: Adam Johnson <notifications@github.com> Date: 19/9/20  9:00 am  (GMT+10:00) To: nopSolutions/nopCommerce <nopCommerce@noreply.github.com> Cc: Subscribed <subscribed@noreply.github.com> Subject: Re: [nopSolutions/nopCommerce] Default Theme - Accessibility - Lack of visual focus indicator (#5022) 

Sure, here's where it's mentioned in the spec and the article where I found it. 👍

**Comment by dylanhthomas** [comment]
Is this an acceptable solution?

**Comment by dylanhthomas** [comment]
@AndreiMaz I have updated the ticket with more information. Hopefully this helps. It aligns with the format and info in the tickets recently submitted by @adamjohnson

Separately, I realized there was a whitespace issue in my PR, so I will be updating that.

**Comment by dylanhthomas** [comment]
Whitespace issues have been cleaned up in the PR
